Add the governance (#2806)
containers/youki: docs/src/community/goversance.md

# Code of Conduct

Youki follows the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md).

# Maintainership

Youki is governed by Maintainers who are elected from active contributors.
Youki will keep the [vendor-neutrality](https://contribute.cncf.io/maintainers/community/vendor-neutrality/).
Maintainers are [here](./maintainer.md).

## Roles

> Maintainers are consist of the following roles:

Reviewer: The user of the term maintainers and committers in the following section is a little confusing. If I understand correctly, maintainers == commiters. If so, I would keep the definition below using the word `commiter`, but standardize on the use of maintainers for the discussion of promotion, demotion, and etc.

Author: @yihuaf @Furisto 
Good point. I was planning to add another Role, Reviewers who are candidates for Comitter. But I didn't write it down then because it would take a little more time to sort out the relevant people.
But, well, I added it because I think it is good to write it down anyway.
https://github.com/containers/youki/pull/2806/commits/09e8cd173d096b919ac497f19fd5871965290442

> Maintainers are consist of the following roles:
> 
> - Committer (Full maintainership)
>     Committers have full write accesses to repos.
>     Committers’ commits should still be made via GitHub pull requests (except for urgent security fixes), and should not be pushed directly.

Reviewer 2: Agree with @yihuaf. If we only have one role for now why do we make the distinction between maintainers and commiters?
```suggestion

- Maintainer
    Maintainer have full write accesses to repos.
    Maintainer's commits should still be made via GitHub pull requests (except for urgent security fixes), and should not be pushed directly.
```
If we decide on more roles in the future then we can make that distinction but for now it is a source of potential confusion because readers will wonder what the difference is.

## Addition and promotion of Maintainers

A contributor who have made significant contributions in quality and in quantity can be also directly invited as a Committer.
A proposal to add or promote a Maintainer must be approved by 2/3 of the Committers who vote within 7 days.
Voting needs 2 approvals at least. The proposer can vote too. Votes from the same company will be counted as one vote.

> A proposal should happen as a GitHub pull request to the Maintainer list above. It is highly suggested to reach out to the Committers before submitting a pull request to check the will of the Committers.

Reviewer: ```suggestion
A proposal should be made via a GitHub pull request to the file containing the list of Maintainer defined above. Before submitting the pull request, the proposer should reach out to the Committers to check their willingness.
```

Reviewer 2: ...to check their willingness to support the proposal.

## Removal and demotion of Maintainers

A Maintainer who do not show significant activities for 12 months, or, who have been violating the Code of Conduct, may be demoted or removed from the project.

A proposal to demote or remove a Maintainer must be approved by 2/3 of the Committers (excluding the person in question) who vote within 14 days.

> Voting needs 2 approvals at least. The proposer can vote too. Votes from the same company will be counted as one vote.

Reviewer: ```suggestion
Voting needs a minimum of 2 approvals. The proposer can vote too. Votes from the same company will be counted as one vote.
```

> A proposal may happen as a GitHub pull request, or, as a private discussion in the case of removal of a harmful Maintainer.

Reviewer: ```suggestion
A proposal should be made via a GitHub pull request to the file containing the list of Maintainer defined above. In the special case of removing a harmful Maintainer, this process can take place via a private discussion.
```

> It is highly suggested to reach out to the Committers before submitting a pull request to check the will of the Committers.

Author: This is a point I had not decided on until now.

Author: Oh, sorry. I meant this is a new decision for our community.

> It is highly suggested to reach out to the Committers before submitting a pull request to check the will of the Committers.

Reviewer: ```suggestion
Before submitting the pull request, the proposer should reach out to the Committers to check their willingness.
```
